### PR TITLE
fix(cmds): option for progress bar in cat/get

### PR DESCRIPTION
**Pull request by schomatis**
Add a `-p=false` flag to not show the progress bar (true by default to keep the current behavior).

Closes https://github.com/ipfs/go-ipfs/issues/8470.

### CODE REVIEW

**Review by guseggert — APPROVED** [review]

File: core/commands/cat.go [hunk 1]
```diff
@@ -96,8 +98,15 @@ var CatCmd = &cmds.Command{
...
-					bar, reader := progressBarForReader(os.Stderr, val, int64(res.Length()))
-					bar.Start()
+					reader := val
+
+					req := res.Request()
+					progress, _ := req.Options[progressOptionName].(bool)
+					if progress {
+						var bar *pb.ProgressBar
+						bar, reader = progressBarForReader(os.Stderr, val, int64(res.Length()))
+						bar.Start()
```
Comment (guseggert) [inline core/commands/cat.go:108]: does this need a `bar.Finish()` somewhere?

Comment (schomatis) [reply core/commands/cat.go:108]: Wouldn't hurt a defer.

**Comment by BigLep** [comment]
@schomatis : any reason we don't merge this given you have approval?

**Comment by schomatis** [comment]
Don't have permissions for the main repo (and I'm fine if we want to keep it this way).

**Pull request merged**